src/Stratis.Bitcoin.Features.LightWallet/LightWalletFeature.cs

```diff
@@ -164,7 +165,9 @@ public override void Dispose()
             this.asyncLoop.Dispose();
             this.walletSyncManager.Stop();
             this.walletManager.Stop();
-            this.lightWalletBlockStoreService.Dispose();
+
+            if (this.storeSettings.PruningEnabled)
```

Comment:
maybe do `this.lightWalletBlockStoreService?.Dispose();` without checking for PruningEnabled, just in case.